src/base/QXmppMessage.cpp

```diff
@@ -962,6 +962,17 @@ void QXmppMessage::setAttachId(const QString &attachId)
     d->attachId = attachId;
 }
 
+///
+/// Returns the participant ID of the sender if the message is received from a MIX channel as
+/// specified in \xep{0369, Mediated Information eXchange (MIX)}.
+///
+/// \since QXmpp 1.7
+///
+QString QXmppMessage::mixParticipantId() const
+{
+    return QXmppUtils::jidToResource(from());
```

Comment:
This should only work with real MIX messages, currently this would return also the resource of normal users